include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
+
+			if (is_array($media['size']))
+				$filesize = intval($media['size'][0]);
+			else
+				$filesize = intval($media['size']);
+
+			if (is_array($media['type']))
+				$filetype = $media['type'][0];
+			else
+				$filetype = $media['type'];
+		}
+
+		if ($filetype=="")
+			$filetype=guess_image_type($filename);
+
+		$imagedata = getimagesize($src);
+		if ($imagedata)
+			$filetype = $imagedata['mime'];
+
+		logger("File upload src: ".$src." - filename: ".$filename.
+			" - size: ".$filesize." - type: ".$filetype, LOGGER_DEBUG);
+
+		// check if there was a php upload error
+		if ($filesize == 0 && $media['error'] == 1)
+			throw new InternalServerErrorException("image size exceeds PHP config settings, file was rejected by server");
+
+		// check against max upload size within Friendica instance
+		$maximagesize = get_config('system','maximagesize');
+		if(($maximagesize) && ($filesize > $maximagesize)) {
+			$formattedBytes = formatBytes($maximagesize);
+			throw new InternalServerErrorException("image size exceeds Friendica config setting (uploaded size: $formattedBytes)");
+		}
+
+		// create Photo instance with the data of the image
+		$imagedata = @file_get_contents($src);
+		$ph = new Photo($imagedata, $filetype);
+		if(! $ph->is_valid()) {
+			throw new InternalServerErrorException("unable to process image data");
+		}
+
+		// check orientation of image
+		$ph->orient($src);
+		@unlink($src);
+
+		// check max length of images on server
+		$max_length = get_config('system','max_image_length');
+		if(! $max_length)
+			$max_length = MAX_IMAGE_LENGTH;
+		if($max_length > 0) {
+			$ph->scaleImage($max_length);
+			logger("File upload: Scaling picture to new size ".$max_length, LOGGER_DEBUG);
+		}
+		$width = $ph->getWidth();
+		$height = $ph->getHeight();
+
+		// create a new resource-id if not already provided
+		$hash = ($photo_id == null) ? photo_new_resource() : $photo_id;
+
+		if ($mediatype == "photo") {
+			// upload normal image (scales 0, 1, 2)
+			logger("photo upload: starting new photo upload", LOGGER_DEBUG);
+
+			$r =$ph->store(local_user(), $visitor, $hash, $filename, $album, 0, 0, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+			if (! $r)
+				logger("photo upload: image upload with scale 0 (original size) failed");
+
+			if($width > 640 || $height > 640) {
+				$ph->scaleImage(640);
+				$r = $ph->store(local_user(),$visitor, $hash, $filename, $album, 1, 0, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+				if (! $r)
+					logger("photo upload: image upload with scale 1 (640x640) failed");
+			}
+
+			if($width > 320 || $height > 320) {
+				$ph->scaleImage(320);
+				$r = $ph->store(local_user(), $visitor, $hash, $filename, $album, 2, 0, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+				if (! $r)
+					logger("photo upload: image upload with scale 2 (320x320) failed");
+			}
+			logger("photo upload: new photo upload ended", LOGGER_DEBUG);
+		}
+		elseif ($mediatype == "profileimage") {
```

Comment:
Standards: Please move the `elseif` on the same line as the preceding closing brace.